subsys/debug/tracing/ctf/ctf_top.c

```diff
@@ -0,0 +1,177 @@
+/*
+ * Copyright (c) 2018 Oticon A/S
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr.h>
+#include <kernel_structs.h>
+#include <init.h>
+
+#include <ctf_middle.h>
+#include "ctf_top.h"
+
+
+#ifndef CONFIG_SMP
+extern k_tid_t const _idle_thread;
+#endif
+
+static inline int is_idle_thread(struct k_thread *thread)
```

Comment:
As far as I remember, this is third copy of `is_idle_thread()` :). Could we make public kernel API from this call?

Reply:
I can do this in this same PR (as a separate commit) if you think it makes sense.

Reply:
Will be done on a later PR.